### PR TITLE
Added the ability to Create a recommendation

### LINKED ISSUE #14: Need ability to Create a recommendation
**As a** manager
 **I need** add a new resource coming
 **So that** I can maintain a database  
   
 ### Details and Assumptions
 * Create a file and then add resources to that file
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given add a resource
 When we call the API
 Then the outcome of action is observed in file
 ```

### PR DESCRIPTION
**Pull request by aianurag09**

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -42,6 +42,21 @@ class Recommendation(db.Model):
...
+        """
+        Creates a Recommendation to the database
+        """
+        # logger.info("Creating %d", self.id)
+        self.id = None  # id must be none to generate next primary key
+        db.session.add(self)
+        db.session.commit()
+    
+    @classmethod
+    def find(cls, id: int):
```
Comment (changtk07) [inline service/models.py]: This find method doesn't belong to this story

File: service/models.py [hunk 2]
```diff
@@ -86,3 +101,10 @@ def init_db(cls, app: Flask):
         db.init_app(app)
         app.app_context().push()
         db.create_all()  # make our sqlalchemy tables
+
+    @classmethod
+    def all(cls) -> list:
```
Comment (changtk07) [inline service/models.py]: This all method doesn't belong to this story neither

File: tests/test_models.py [hunk 3]
```diff
@@ -61,7 +61,6 @@ def test_serialize_a_recommendation(self):
         self.assertIn("rec_product_id", data)
         self.assertEqual(data["rec_product_id"], recommendation.rec_product_id)
         self.assertIn("type", data)
-        self.assertEqual(data["type"], recommendation.type.name)
```
Comment (changtk07) [inline tests/test_models.py]: Is there a reason to delete this line?

File: tests/test_routes.py [hunk 4]
```diff
@@ -66,6 +66,22 @@ def _create_recommendations(self, count):
             test_recommendation.id = new_recommendation["id"]
             recommendations.append(test_recommendation)
         return recommendations
+    
+    def _create_recommendations(self, count):
```
Comment (changtk07) [inline tests/test_routes.py]: Duplicate function

**Review by changtk07 — CHANGES_REQUESTED** [review]
Reviews

File: tests/test_models.py (hunk 3 above)
Comment (aianurag09) [reply tests/test_models.py]: I should have removed it by mistake will add that line

**Review by changtk07 — APPROVED** [review]
Approved

**Comment by aianurag09** [comment]
Shall I merge @changtk07, so that other can work on this edits

**Comment by changtk07** [comment]
Does their functions rely on your codes?

**Comment by aianurag09** [comment]
Yes it will

**Comment by aianurag09** [comment]
@changtk07

**Comment by changtk07** [comment]
Then maybe they should clone from your branch and work along @aianurag09

**Comment by aianurag09** [comment]
It will again create a lot of issues I guess, you already reviewed it and we can again check if there are any issues right?

**Pull request merged**